src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -85,7 +85,8 @@ public String toString() {
         public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item project,
                                                      @QueryParameter String url,
                                                      @QueryParameter String credentialsId) {
-            if (project == null || !project.hasPermission(Item.EXTENDED_READ)) {
+            if (project == null && !Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER) ||
+                project != null && !project.hasPermission(Item.EXTENDED_READ)) {
```

Comment:
This is the diff hunk currently covered by the functional test—the one most important for the reported bug.